identifier/vite.config.js

```diff
@@ -1,6 +1,41 @@
-import { defineConfig, splitVendorChunkPlugin } from 'vite';
-import react from '@vitejs/plugin-react';
-import checker from 'vite-plugin-checker';
+import { defineConfig, splitVendorChunkPlugin } from "vite";
+import react from "@vitejs/plugin-react";
+import checker from "vite-plugin-checker";
+import legacy from "@vitejs/plugin-legacy";
+import { readFileSync, writeFileSync } from "fs";
+import path from "path";
+
+function cspPlugin() {
+  return {
+    name: "csp-plugin",
+    closeBundle() {
+      // Path to the generated HTML file after build
+      const htmlFilePath = path.resolve(__dirname, "build/index.html");
+
+      // Read the HTML file content
+      let htmlContent = readFileSync(htmlFilePath, "utf-8");
+
+      // Insert the CSP meta tag in the HTML <head> section
+      htmlContent = htmlContent.replaceAll(
+        /<script nomodule>/gi,
+        `<script nomodule content="__CSP_NONCE__">`
```

Comment:
This is wrong and does not work (wrong attribute `content`) - please make sure to test things before submitting changed PRs as this seems impossible to be functional.

Reply:
Updated this with nonce.